shoes-swt/lib/shoes/swt/image.rb

```diff
@@ -77,10 +78,18 @@ def default_height
       end
 
       def download_and_display_real_image(url)
-        @tmpname = File.join(Dir.tmpdir, "__shoes4_#{Time.now.to_f}.png") unless @tmpname_or_data
+        if @is_image
```

Comment:
I agree there's an opportunity there. I'd be most in favor of actually having a separate pair of classes -- `ImageDisplayer` and `GifDisplayer` maybe?-- that we use within `Shoes::Swt::Image`.

A subclass of `Shoes::Swt::Image` would be nice in some ways, but it fights against the structure of how the rest of Shoes wants to work DSL class -> Backend class.